### DISCUSSION TITLE
My laser setup

### DISCUSSION THREAD

**Discussion by mundsen** [phil-barrett/grblHAL-teensy-4.x#112] [Show and tell]
![436549528_3710616465919007_7636087309870444278_n (1)](https://github.com/user-attachments/assets/fc0d2b20-a20b-4cb8-b803-bbefc83e0355)
I have used controller from Phil in all my laser projects - love it. 

This is the setup I use at the moment, but I`m going to do some upgrades.

I`m working on a laser project where we are testing what speeds.. are possible

Upgrade plans:

- Upgrade from 36V to 56V stepper controller power
- DM542 Stepper drivers
- a better relay module (use it to control my exhaust/smoke fan..)

**Comment by phil-barrett** [comment]
That is really nice.

**Comment by malcolmmp** [comment]
Hi @mundsen! Looks great!

I just noticed that it looks like you're utilizing quick disconnect screw terminals. Where did you end up sourcing those?

I'm looking to build another machine soon with the same Teensy controller, and I was coming up short looking attempting to find the headers with the correct pitch. I've got a couple of Mesa cards for LinuxCNC and it looks like they're using what I'm looking for, though the board mounted headers have a different overall number of sockets. Not sure if 24 position is any easier to source than the 20 and 22s that would be required on the Teensy controller. The headers on those have the sockets coming out parallel to the plane of the board, rather than perpendicular, if that make sense.

@phil-barrett have you looked into those kind of connectors for your boards at all and if so, do you have any leads on where to source them?

**Reply by malcolmmp** [reply]
![IMG_1154](https://github.com/user-attachments/assets/e1be9957-939d-4b97-855d-3cd797e81d85)
Image for reference

**Comment by phil-barrett** [comment]
Mundson got them from me. I got them from LCSC in China.  US and EU distis have them but charge amazingly high prices. Buying from china got the cost down but shipping is expensive.  Even so, it was still cheaper. Unfortunately, LCSC doesn't make it easy to figure out what works together.

You have to order the base and plug-in parts together to ensure a good fit. I found it impossible to figure out what plug-in part the base needed by looking at the base. It was one of the reasons the design of the pro board became troublesome.  Take a hard pass if you ever see a board or product for sale that uses pluggable screw terminals but doesn't come with the plug-in parts.

**Reply by mundsen** [reply]
In the grblhal facebook group, there is a guy (Richard Toth) that use screw terminals like this
![340464637_749362416797201_6989718839048255875_n](https://github.com/user-attachments/assets/493cd418-3ced-47c7-8f5f-f0cac29bb476)


https://www.soselectronic.com/sk/products/euroclamp/pv02-3-5-v-75875

https://www.soselectronic.com/sk/products/euroclamp/sh02-3-5-51473

But I have not tested them, so I do not know if they have the same problems as Phil describes

**Reply by phil-barrett** [reply]
If you buy the base and the plug-in together, it should be fine.

**Reply by mundsen** [reply]
![koblinger](https://github.com/user-attachments/assets/63d69944-e205-4982-a745-d99522af22d4)

Seems like this is the minimal order for one board - a bit expensive - this is without shipping, tax..

**Reply by mundsen** [reply]
Better if selecting only one color

![koblinger2](https://github.com/user-attachments/assets/80121c2b-e21f-4957-855b-21dcb8815945)

**Reply by malcolmmp** [reply]
Thanks for the info! I'll definitely be picking those up when I start building the next machine!

If memory serves, I found equivalents to what the Mesa boards have, but prices would have put just the headers at $50+USD on Digikey with multiple months of lead time, so it was a non-starter.

I really like the way the ones above are modular so you're not stuck sourcing specific pin number configs.

**Reply by phil-barrett** [reply]
I would just use 2 pin connectors and not try to mix 2 and 4 pin ones.  It's not as nice for things like stepper/servo drivers but you can order in higher volume (to maybe get a price cut) and they are all interchangeable.

**Comment by mundsen** [comment]
For my air assist I use an solenoid valve that I set on or off for each layer in Lightburn - using the relay module
https://github.com/user-attachments/assets/98d73135-57cc-4f53-a11d-4975e2e1a915
